service/dap/server.go

```diff
@@ -451,6 +451,43 @@ func (s *Server) onContinueRequest(request *dap.ContinueRequest) {
 	s.doContinue()
 }
 
+func (s *Server) onThreadsRequest(request *dap.ThreadsRequest) {
+	if s.debugger == nil {
+		s.sendErrorResponse(request.Request, UnableToDisplayThreads, "Unable to display threads", "debugger is nil")
+		return
+	}
+	gs, _, err := s.debugger.Goroutines(0, 0)
+	if err != nil {
+		switch err.(type) {
+		case *proc.ErrProcessExited:
+			// If the program exits very quickly, the initial threads request will complete after it has exited.
+			// A TerminatedEvent has already been sent. Ignore the err returned in this case.
+			s.send(&dap.ThreadsResponse{Response: *newResponse(request.Request)})
+		default:
+			s.sendErrorResponse(request.Request, UnableToDisplayThreads, "Unable to display threads", err.Error())
+		}
+		return
+	}
+
+	threads := make([]dap.Thread, len(gs))
```

Comment:
Nit: this would be clearer to read if the cases would be more delineated, e.g.:

```
var threads []dap.Thread
if len(threads) == 0 {
  threads = ... dummy
} else {
  // fill threads from gs
}
response := ... {Threads: threads}
```

Right now it's a bit harder to follow, at least for me.

Up to you.